src/db/collections/block.rs

```diff
@@ -264,23 +265,35 @@ impl BlockCollection {
         &self,
         index: MilestoneIndex,
     ) -> Result<impl Stream<Item = Result<(BlockId, Block, Vec<u8>, BlockMetadata), Error>>, Error> {
+        #[derive(Debug, Deserialize)]
+        struct QueryRes {
+            #[serde(rename = "_id")]
+            block_id: BlockId,
+            #[serde(with = "serde_bytes")]
+            raw: Vec<u8>,
+            metadata: BlockMetadata,
+        }
+
         Ok(self
-            .aggregate::<BlockDocument>(
+            .aggregate::<QueryRes>(
                 vec![
                     doc! { "$match": { "metadata.referenced_by_milestone_index": index } },
                     doc! { "$sort": { "metadata.white_flag_index": 1 } },
                 ],
                 None,
             )
             .await?
-            .map_ok(
-                |BlockDocument {
-                     block_id,
-                     block,
-                     raw,
-                     metadata,
-                 }| (block_id, block, raw, metadata),
-            ))
+            .map_ok(|r| {
+                (
+                    r.block_id,
+                    RawMessage::<iota_types::block::Block>::from(r.raw.clone())
+                        .inner_unverified()
+                        .unwrap()
+                        .into(),
```

Comment:
Can you help me understand this better? So ... eventually we need the protocol parameters from the time where this `Block` was serialized and written to the db?

Reply:
No, the problem is we don't store `Output`s in the `BlockCollection`

Reply:
ah, I see, thanks